sorting/external_sort.cpp

```diff
@@ -0,0 +1,187 @@
+/**
+ * C++ program to sort a file data using external sorting
+ * External sorting is required when the data being sorted do not fit into the
+ * memory (RAM)
+ */
+
+#include <algorithm>
+#include <cassert>
+#include <fstream>
+#include <iostream>
+#include <random>
+#include <string>
+
+using namespace std;
+
+// buff data which fit in memory to sort
+constexpr auto BUFF_SIZE = 100000;
+
+struct ChunkFile {
+    ifstream file;
+    int currentVal;
+    string line;
+};
+
+/**
+   Prepare 1M rows data (int) data being sorted and write to text file.
+ */
+void mockData() {
+    cout << "Preparing data" << endl;
+
+    // change to absolutely path if u want
+    ofstream data_file("data.txt", ofstream::trunc);  // overwite
+
+    // use to get a random number
+    mt19937 generator(time(0));
+    uniform_int_distribution<int> distribution(0, 999999);
+
+    for (int i = 0; i < 1000000; i++) {
+        // do random number
+        int randomNumber = distribution(generator);
+        data_file << randomNumber << "\n";
+    }
+    data_file.close();
+
+    cout << "End preparing data" << endl;
+}
```

Comment:
```suggestion
void mockData(char* path = "data.txt") {
    ofstream data_file(path , ofstream::trunc);  // overwite

    // use to get a random number
    mt19937 generator(time(0));
    uniform_int_distribution<int> distribution(0, 999999);

    for (int i = 0; i < 1000000; i++) {
        int randomNumber = distribution(generator);
        data_file << randomNumber << "\n";
    }
    data_file.close();
}
```
wouldnt it be better to have a path parameter to specify the path of the file